File: src/com/prey/json/actions/Wifi.java
package com.prey.json.actions;

import java.util.HashMap;
import java.util.List;

import org.json.JSONObject;

import android.content.Context;

import com.prey.PreyLogger;
import com.prey.PreyPhone;
import com.prey.actions.HttpDataService;
import com.prey.actions.observer.ActionResult;
import com.prey.json.JsonAction;

public class Wifi extends JsonAction {
	
	public static String SSID="ssid";

	@Override
	public HttpDataService run(Context ctx, List<ActionResult> lista, JSONObject parameters) {
		HttpDataService data = new HttpDataService("wifi");
		try {

			data.setList(true);

			PreyPhone phone = new PreyPhone(ctx);
			com.prey.PreyPhone.Wifi wifiPhone = phone.getWifi();

			HashMap<String, String> parametersMap = new HashMap<String, String>();

		 

			parametersMap.put(SSID, wifiPhone.getSsid());
			parametersMap.put("mac_address", wifiPhone.getMacAddress());
			parametersMap.put("security", wifiPhone.getSecurity());
			parametersMap.put("signal_strength", wifiPhone.getSignalStrength());
			parametersMap.put("channel", wifiPhone.getChannel());

			parametersMap.put("interfaceType", wifiPhone.getInterfaceType());
			parametersMap.put("model", wifiPhone.getModel());
			parametersMap.put("vendor", wifiPhone.getVendor());
			parametersMap.put("ipAddress", wifiPhone.getIpAddress());
			parametersMap.put("gatewayIp", wifiPhone.getGatewayIp());
			parametersMap.put("netmask", wifiPhone.getNetmask());

			data.addDataListAll(parametersMap);

		} catch (Exception e) {
			PreyLogger.e("Error causa:" + e.getMessage() + e.getMessage(), e);
		}
		return data;
	}

}

File: app/src/main/java/com/prey/json/actions/kotlin/Wifi.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.json.actions.kotlin

import android.content.Context
import com.prey.actions.kotlin.HttpDataService
import com.prey.actions.observer.kotlin.ActionResult
import com.prey.kotlin.PreyLogger
import com.prey.kotlin.PreyPhone

import org.json.JSONObject

class Wifi {
    fun run(
        ctx: Context,
        list: List<ActionResult>?,
        parameters: JSONObject?
    ): HttpDataService {
        val data = HttpDataService("wifi")
        try {
            data.setList(true)
            val phone = PreyPhone(ctx)
            val wifiPhone = phone.wifi!!
            val parametersMap = HashMap<String, String?>()
            parametersMap[SSID] = wifiPhone.ssid!!
            parametersMap["mac_address"] = wifiPhone.macAddress!!
            parametersMap["security"] = wifiPhone.security!!
            parametersMap["signal_strength"] = wifiPhone.signalStrength!!
            parametersMap["channel"] = wifiPhone.channel!!
            parametersMap["interfaceType"] = wifiPhone.interfaceType!!
            parametersMap["model"] = wifiPhone.model!!
            parametersMap["vendor"] = wifiPhone.vendor!!
            parametersMap["ipAddress"] = wifiPhone.ipAddress!!
            parametersMap["gatewayIp"] = wifiPhone.gatewayIp!!
            parametersMap["netmask"] = wifiPhone.netmask!!
            data.addDataListAll(parametersMap)
        } catch (e: Exception) {
            PreyLogger.e("Error:" + e.message + e.message, e)
        }
        return data
    }

    companion object {
        var SSID: String = "ssid"
    }
}